django/contrib/admin/options.py

```diff
@@ -213,6 +217,14 @@ def formfield_for_foreignkey(self, db_field, request, **kwargs):
         Get a form Field for a ForeignKey.
         """
         db = kwargs.get('using')
+
+        if db_field.name in self.autocomplete_fields:
+            related_admin = self.admin_site._registry.get(db_field.remote_field.model)
+            if not related_admin.search_fields:
+                msg = 'The admin for %s needs to implement "search_fields".'
+                raise NotImplementedError(msg % db_field.remote_field.model)
+            kwargs['widget'] = AutocompleteSelect(model_admin=self, using=db)
+
         if db_field.name in self.raw_id_fields:
```

Comment:
change "if" to "elif" -- the field can't be in autocomplete_fields and raw_id_fields, correct?